### PR TITLE
[Bigtable] Table and column family admin APIs

### PR DESCRIPTION
**Pull request by jiren**
- Table create, delete and list APIs
- Column families create, update, delete and bulk modification
- Table and Column families mock unit and acceptance test cases
- Refactored longrunning job, remove redundant job methods and move into base job class `LongrunningJob`

### CODE REVIEW

File: google-cloud-bigtable/lib/google/cloud/bigtable/gc_rule.rb [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+module Google
+  module Cloud
+    module Bigtable
+      # # GcRule
+      #
+      # Rule for determining which cells to delete during garbage collection
```
Comment (quartzmo) [inline google-cloud-bigtable/lib/google/cloud/bigtable/gc_rule.rb]: missing period at end of line

File: google-cloud-bigtable/lib/google/cloud/bigtable/gc_rule.rb [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+      # NOTE: GC Rule can hold only one type at a time.
+      # GC Rule types:
+      #   * `max_num_versions` - Delete all cells in a column except the most recent N
+      #   * `max_age` - Delete cells in a column older than the given age.
+      #   * `union` - Delete cells that would be deleted by every nested rule.
+      #       It can have mutiple chainable GC Rules.
+      #   * `intersection` - Delete cells that would be deleted by any nested rule.
+      #       It can have mutiple chainable GC Rules.
+      #
+      # @example Create gc rule instance with max version.
```
Comment (quartzmo) [inline google-cloud-bigtable/lib/google/cloud/bigtable/gc_rule.rb]: Capitalization/spelling here should be "GC Rule".

File: google-cloud-bigtable/lib/google/cloud/bigtable/gc_rule.rb [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+      #
+      #  gc_rule = Google::Cloud::Bigtable::GcRule.max_versions(3)
+      #
+      #  # Change max verions
+      #  gc_rule.max_versions = 5
+      #
+      # @example Create gc rule instance with max age.
```
Comment (quartzmo) [inline google-cloud-bigtable/lib/google/cloud/bigtable/gc_rule.rb]: Capitalization/spelling here should be "GC Rule".

File: google-cloud-bigtable/lib/google/cloud/bigtable/gc_rule.rb [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+      #
+      #  gc_rule = Google::Cloud::Bigtable::GcRule.max_age(3)
+      #
+      #  # Change max age
+      #  gc_rule.max_age = 600 # 10 minutes
+      #
+      # @example Create gc rule instance with union.
```
Comment (quartzmo) [inline google-cloud-bigtable/lib/google/cloud/bigtable/gc_rule.rb]: Capitalization/spelling here should be "GC Rule".

File: google-cloud-bigtable/lib/google/cloud/bigtable/gc_rule.rb [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+      #
+      #  max_age_gc_rule = Google::Cloud::Bigtable::GcRule.max_age(180)
+      #  union_gc_rule = Google::Cloud::Bigtable::GcRule.union(max_age_gc_rule)
+      #
+      #  # Change union gc rule
```
Comment (quartzmo) [inline google-cloud-bigtable/lib/google/cloud/bigtable/gc_rule.rb]: Capitalization/spelling here should be "GC Rule".

File: google-cloud-bigtable/lib/google/cloud/bigtable/gc_rule.rb [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+      #  gc_rule.union = Google::Cloud::Bigtable::GcRule.max_age(600)
+      #
+      # @example Create gc rule instance with intersection.
```
Comment (quartzmo) [inline google-cloud-bigtable/lib/google/cloud/bigtable/gc_rule.rb]: Capitalization/spelling here should be "GC Rule".

File: google-cloud-bigtable/lib/google/cloud/bigtable/gc_rule.rb [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+      #
+      #  max_versions_gc_rule = Google::Cloud::Bigtable::GcRule.max_versions(3)
+      #  gc_rule = Google::Cloud::Bigtable::GcRule.intersection(max_versions_gc_rule)
+      #
+      #  # Change intersection gc rule
```
Comment (quartzmo) [inline google-cloud-bigtable/lib/google/cloud/bigtable/gc_rule.rb]: Capitalization/spelling here should be "GC Rule".

File: google-cloud-bigtable/lib/google/cloud/bigtable/gc_rule.rb [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+        #   List of GcRule with nested rules.
+        #
+        def intersection= rules
+          @grpc.intersection = \
+            Google::Bigtable::Admin::V2::GcRule::Intersection.new(
+              rules: rules.map(&:to_grpc)
+            )
+        end
+
+        # Get intersection gc rules
```
Comment (quartzmo) [inline google-cloud-bigtable/lib/google/cloud/bigtable/gc_rule.rb]: Capitalization/spelling here should be "GC Rules".

File: google-cloud-bigtable/lib/google/cloud/bigtable/gc_rule.rb [hunk 9]
```diff
@@ -0,0 +1,215 @@
...
+        # @param rules [Array<Google::Cloud::Bigtable::GcRule>]
+        #   List of GcRule with nested rules.
+        #
+        def union= rules
+          @grpc.union = Google::Bigtable::Admin::V2::GcRule::Union.new(
+            rules: rules.map(&:to_grpc)
+          )
+        end
+
+        # Get union gc rules
```
Comment (quartzmo) [inline google-cloud-bigtable/lib/google/cloud/bigtable/gc_rule.rb]: Capitalization/spelling here should be "GC Rules".

**Comment by sduskis** [comment]
@blowmage, I'm ok with this being merged.  Once all of the pull requests are done, I'll do a thorough review before we mark this client as alpha

**Pull request merged**